lib/build-cloud.rb

```diff
@@ -220,16 +223,31 @@ def recursive_interpolate_config(h)
         when String
 
             while h =~ /%\{(.+?)\}/
-                var = $1
+                exp = $1
+
+                var = ""
                 val = ""
+                default = ""
+
+                if exp =~ /\|\|/
+                    (var, default) = exp.split('||')
+                else
+                    var = exp
+                end
 
                 if @config.has_key?(var.to_sym)
```

Comment:
Can probably do some magic with `#fetch` here. I think you'd be able to remove at least some of this `if` block by using it. At the very least `has_key?` is deprecated. `key?` is your new friend.